src/Enum/Setting.php

```diff
@@ -0,0 +1,123 @@
+<?php
+
+/**
+ * Copyright © 2019-present Spryker Systems GmbH. All rights reserved.
+ * Use of this software requires acceptance of the Evaluation License Agreement. See LICENSE file.
+ */
+
+namespace SprykerSdk\SdkContracts\Enum;
+
+final class Setting
+{
+    /**
+     * Type for shared settings in repository.
+     *
+     * @var string
+     */
+    public const SETTING_TYPE_SHARED = 'shared';
+
+    /**
+     * Type for local settings.
+     *
+     * @var string
+     */
+    public const SETTING_TYPE_LOCAL = 'local';
+
+    /**
+     * Type for sdk settings in repository.
+     *
+     * @var string
+     */
+    public const SETTING_TYPE_SDK = 'sdk';
+
+    /**
+     * Path setting key for extension directories.
+     *
+     * @var string
+     */
+    public const PATH_EXTENSION_DIRS = 'extension_dirs';
+
+    /**
+     * Path setting key for sdk directory.
+     *
+     * @var string
+     */
+    public const PATH_SDK_DIR = 'sdk_dir';
+
+    /**
+     * Path setting key for project directory.
+     *
+     * @var string
+     */
+    public const PATH_PROJECT_DIR = 'project_dir';
+
+    /**
+     * Path setting key for report usage statistic.
+     *
+     * @var string
+     */
+    public const PATH_REPORT_USAGE_STATISTICS = 'report_usage_statistics';
+
+    /**
+     * Path setting key for core namespaces.
+     *
+     * @var string
+     */
+    public const PATH_CORE_NAMESPACES = 'coreNamespaces';
+
+    /**
+     * Path setting key for report project namespaces.
+     *
+     * @var string
+     */
+    public const PATH_PROJECT_NAMESPACES = 'projectNamespaces';
+
+    /**
+     * Path setting key for default violation format.
+     *
+     * @var string
+     */
+    public const PATH_DEFAULT_VIOLATION_OUTPUT_FORMAT = 'default_violation_output_format';
```

Comment:
I am a bit confused about the different casings
camelBacked 'projectNamespaces' vs underscored 'default_violation_output_format'

I feel like this should be harmonized if you make this now an explicit cross-repo dependency for long term constant usage.